src/main/java/org/opensearch/knn/index/mapper/LuceneFieldMapper.java

```diff
@@ -92,7 +92,9 @@ protected void parseCreateField(ParseContext context, int dimension, SpaceType s
             KnnByteVectorField point = new KnnByteVectorField(name(), array, fieldType);
 
             context.doc().add(point);
-            addStoredFieldForVectorField(context, fieldType, name(), point);
+            if (this.stored) {
+                context.doc().add(createStoredFieldForVectorField(name(), array));
+            }
```

Comment:
can we use the older abstraction where we were validating if we need to store the vector field or not. Rather than adding a if condition here?

Reply:
The old abstraction ended up breaking because in our parameter, in our field mapper build store parameter, we set the stored value for the [KNNFieldMapper parameter](https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapper.java#L112) instead of the lucene FieldType parameter:
```
protected final Parameter<Boolean> stored = Parameter.boolParam("store", false, m -> toType(m).stored, false);
```

So, with old abstraction, we would either need to modify it to
1. pass in the KNNVectorFieldMapper object
2. pass in the `bool stored` parameter
3. modify the Lucene FieldType to set stored there.

In terms of setting at [KNNVectorFieldMapper](https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapper.java#L489) or [FieldType](https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/document/FieldType.java#L114-L117) level, Im not sure what is best practice. Ive seen it done both ways:
1. [TextFieldMapper](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/index/mapper/TextFieldMapper.java#L284)
2. [NumberFieldMapper](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/index/mapper/NumberFieldMapper.java#L1510)

@navneet1v do you have any opinions on this? Im going to dig a little bit more in Lucene to see if there are any pitfalls of not setting FieldType.stored but I dont see any obvious ones.

I can switch back to the old abstraction, but then Id probably rename from `addStoredFieldForVectorField` to `maybeAddStoredFieldForVectorField` so that it isnt misleading.

Reply:
> Im going to dig a little bit more in Lucene to see if there are any pitfalls of not setting FieldType.stored but I dont see any obvious ones.

https://github.com/apache/lucene/blob/3024e66e4aba942b039fcad7daf958aa4c90b8bf/lucene/core/src/java/org/apache/lucene/document/Field.java#L606-L625. I looked into this, I dont think its applicable. I think storing in the KNNFieldMapper parameter is fine. @navneet1v that being said, I can change `addStoredFieldForVectorField` -> `maybeAddStoredFieldForVectorField` and pass in the bool stored parameter to remove some redundant "if statements" in the code. Im hesistant to pass in the KNNVectorFieldMapper because I think it has a lot more functionality than is required for the helper method.

Reply:
>  Im hesistant to pass in the KNNVectorFieldMapper because I think it has a lot more functionality than is required for the helper method.


I agree on this.

I think now we have overloaded the functions the code bit more clean. Having if statements is fine here.